pkg/planner/core/rule_column_pruning.go

```diff
@@ -205,6 +206,55 @@ func (la *LogicalAggregation) PruneColumns(parentUsedCols []*expression.Column,
 			la.SetChildren(childOfChild)
 		}
 	}
+	// Check for Apply elimination
+	if err := la.tryEliminateApply(opt); err != nil {
```

Comment:
Column pruning is a top-down optimization process, so what should be changed is to expand logicalapply's PruneColumns() instead of adding a new tree traversal logic at the end of aggregation's PruneColumns.